File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let n = nums.length;
    let ans = new Array(n);

    let left = 1;
    for (let i = 0; i < n; i++) {
        ans[i] = left;
        left = left * nums[i];
    }

    let right = 1;
    for (let i = n - 1; i >= 0; i--) {
        ans[i] = ans[i] * right;
        right = right * nums[i];
    }
    return ans;
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    var n int = len(nums)
    var ans []int= make([]int, n)

    var left int = 1
    for i := 0; i < n; i++ {
        ans[i] = left
        left = left * nums[i]
    }

    var right int = 1
    for i := n - 1; i >= 0; i-- {
        ans[i] = ans[i] * right
        right = right * nums[i]
    }

    return ans
}